Sprint 9:

ihatelags opened the pull request:
- feat: changed to layered architecture
- feat: added likes and friends
- feat: improved REST

ihatelags committed:
Sprint 9:
- feat: changed to layered architecture
- feat: added likes and friends
- feat: improved REST

ihatelags committed:
validation fixes according to Postman tests

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -18,4 +22,14 @@ public class Film {
     @Min(1)
     private int duration;
+    @JsonIgnore
+    private HashSet<Long> likes;
```
объявлять лучше интерфейсом Set

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -17,4 +21,6 @@ public class User {
     @Past
     private LocalDate birthday;
+    @JsonIgnore
+    private HashSet<Long> friends;
```
объявлять лучше интерфейсом Set

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/storage/UserStorage.java:
```diff
@@ -0,0 +1,24 @@
+import java.util.List;
+
+public interface UserStorage {
+    HashMap<Long, User> getAllUsers();
```
объявлять лучше интерфейсом Map

utausheva replied:
а вообще возвращать нужно не хэшмапу, а список)

utausheva replied:
как в фильмах

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryFilmStorage.java:
```diff
@@ -0,0 +1,63 @@
+@Slf4j
+@Component
+public class InMemoryFilmStorage implements FilmStorage {
+    private final HashMap<Long, Film> films = new HashMap<>();
```
объявлять лучше интерфейсом Map

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryUserStorage.java:
```diff
@@ -0,0 +1,108 @@
+@Slf4j
+@Component
+public class InMemoryUserStorage implements UserStorage {
+    private final HashMap<Long, User> users = new HashMap<>();
```
объявлять лучше интерфейсом Map

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryUserStorage.java:
```diff
@@ -0,0 +1,108 @@
+public class InMemoryUserStorage implements UserStorage {
+    private final HashMap<Long, User> users = new HashMap<>();
+
+    public HashMap<Long, User> getAllUsers() {
```
здесь подправить возвращаемые данные

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryUserStorage.java:
```diff
@@ -0,0 +1,108 @@
+        HashSet<Long> friendsOfUser;
+        HashSet<Long> friendsOfAnotherUser;
+
+        if (id != null && otherId != null) {
```
попробуй стримом, пишется в 3 строки))

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,114 @@
+
+    public List<Film> getPopularFilms(Integer count) {
+        List<Film> films = filmStorage.getPopularFilms(count);
+        return films.stream().limit(Objects.requireNonNullElse(count, 10)).collect(Collectors.toList());
```
у тебя прописано дефолтное значение count в контроллере, здесь можно не писать еще раз 10) А то в случае изменения дефолтного лимита придется править в двух местах вместо одного

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/controller/ErrorHandler.java:
```diff
@@ -0,0 +1,34 @@
+
+    @ExceptionHandler
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ErrorResponse handleValidationException(final ValidationException validationException) {
```
Для возврата сообщения, обернутого в ответ со статусом можно использовать ResponseEntity, таким образом мы избавимся от создания лишнего класса и сможем убрать аннотацию @ResponseStatus. Возвращаемый тип будет ResponseEntity<?>, а возвратом будет ResponseEntity<>(сообщение, статус)
(не обязательно)

utausheva reviewed (commented):
Привет!
Очень круто, что не забыл ни одного модификатора доступа)))
Есть незначительные замечания, подправляй и можно принимать)

ihatelags committed:
- fix: заменил объявление переменных на интерфейсы
- fix: переписал getCommonFriends
- fix: обновил ErrorHandler на использование ResponseEntity

ihatelags committed:
- fix: заменил объявление переменных на интерфейсы
- fix: переписал getCommonFriends
- fix: обновил ErrorHandler на использование ResponseEntity

ihatelags committed:
Merge remote-tracking branch 'origin/add-friends-likes' into add-friends-likes

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryUserStorage.java

ihatelags committed:
- fix: добавил try/catch в getCommonFriends

utausheva reviewed (approved):
Супер!
Метод getCommonFriends огонь просто)
Удачи на SQL)))
